plugins/view-resources/src/components/Menu.svelte

```diff
@@ -41,37 +41,47 @@
     other: 6,
     remove: 7
   }
+  const findAllObjects = async (object: Doc | Doc[]) => {
+    const objArray = Array.isArray(object) ? object : [object]
+    const objClass = objArray.map((obj) => obj._class).filter((it, idx, arr) => arr.indexOf(it) === idx)
+    if (objClass.length === 1) {
+      return await client.findAll(objClass[0], { _id: { $in: objArray.map((it) => it._id) } })
+    }
+    return object
+  }
 
-  void getActions(client, object, baseMenuClass, mode).then((result) => {
-    const filtered = result.filter((a) => {
-      if (excludedActions.includes(a._id)) {
-        return false
-      }
-      if (includedActions.length > 0 && !includedActions.includes(a._id)) {
-        return false
+  void findAllObjects(object).then((res) => {
```

Comment:
Should we show progress somehow?

Reply:
Yes, I'm working on uitests, will rewrite this so it could load better